j5/backends/hardware/sr/v4/power_board.py

```diff
@@ -65,6 +66,41 @@ class WriteCommand(NamedTuple):
 CMD_WRITE_PIEZO = WriteCommand(8)
 
 
+def handle_usb_error(func):
+    """
+    Wrap functions that use usb1 and give friendly errors.
+
+    The exceptions from usb1 are hard to find in documentation or code and are confusing
+    to users. This decorator catches the USBErrors and throws a friendlier exception that
+    can also be caught more easily.
+    """
+
+    @wraps(func)
+    def catch_exceptions(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except usb1.USBErrorNoDevice:
+            raise CommunicationError("USB Device not found. Is it connected?") from None
+        except (
+                usb1.USBErrorIO,
+                usb1.USBErrorInvalidParam,
+                usb1.USBErrorAccess,
+                usb1.USBErrorNotFound,
+                usb1.USBErrorBusy,
+                usb1.USBErrorTimeout,
+                usb1.USBErrorOverflow,
+                usb1.USBErrorPipe,
+                usb1.USBErrorInterrupted,
+                usb1.USBErrorNoMem,
+                usb1.USBErrorNotSupported,
+                usb1.USBErrorOther,
+
+        ) as e:
+            name = usb1.libusb1.libusb_error.get(e.value, 'Unknown Error.')
+            raise CommunicationError(f"USB Error({name})") from None
```

Comment:
Please can we not use `from None` here? It makes it harder to find the underlying cause of the error, which sometimes may require delving into the source of `usb1` if the error messages provided by that library are sufficiently unhelpful.